src/h5web/visualizations/nexus/utils.ts

```diff
@@ -124,15 +125,49 @@ export function getLinkedDatasets(
   return Object.fromEntries(datasetEntries);
 }
 
-export function getNxAxesParams(
-  nxDatasets: Record<string, HDF5Dataset>,
-  datasetValues: Record<string, HDF5Value> | undefined
-): Record<string, AxisParams> {
-  const paramsEntries = Object.entries(nxDatasets).map(([axis, dataset]) => {
-    const values = datasetValues && datasetValues[axis];
-    assertOptionalArray<number>(values);
-    return [axis, { values, label: getDatasetLabel(dataset, axis) }];
+export function useParsedNxData(
+  group: HDF5Group,
+  metadata: HDF5Metadata
+): NxData | undefined {
+  const datasetValues = useChildDatasetValues(group, metadata);
+  if (!datasetValues) {
+    return undefined;
+  }
+
+  const signalName = getAttributeValue(group, 'signal');
+  assertStr(signalName);
+  const signalDataset = getLinkedEntity(signalName, group, metadata);
+  const signalValue = datasetValues[signalName];
+
+  if (!signalDataset || !isDataset(signalDataset) || !signalValue) {
+    return undefined;
```

Comment:
I think this is where you should throw `'Signal dataset not found'`.

Reply:
You can also remove `!isDataset()` from the check and use `assertDataset()` instead.